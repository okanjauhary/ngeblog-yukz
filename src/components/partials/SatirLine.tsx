import React, { useState, useEffect } from "react";
import { css } from "@emotion/core";
import { getTheme } from "../../utils/theme";

/*
 * a line divider
 * "Satir" is divider for each part of room or people groups in java culture
 */

const SatirLine: React.FC = () => {
  const [lineColor, setLineColor] = useState("#000");

  useEffect(() => {
    setLineColor(getTheme("--text-color"));
  });

  return (
    <div
      className="satir-line"
      css={css`
        margin: 10px 0;
        display: flex;
        align-items: center;

        &::before {
          content: "";
          display: block;
          width: 100%;
          height: 2px;
          background-color: ${lineColor};
        }
      `}></div>
  );
};

export default SatirLine;
