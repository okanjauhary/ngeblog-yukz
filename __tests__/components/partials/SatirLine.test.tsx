import React from "react";
import { mount } from "enzyme";
import SatirLine from "../../../src/components/partials/SatirLine";

describe("SatirLine.tsx", () => {
  test("Should render component and match snapshot", () => {
    const wrapper = mount(<SatirLine />);

    expect(wrapper).toMatchSnapshot();
  });
});
