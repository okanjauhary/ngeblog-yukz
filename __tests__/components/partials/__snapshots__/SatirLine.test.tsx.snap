// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays the correct title 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="satir-line"
        css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="satir-line"
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
